#!/usr/bin/env python
PACKAGE = "ros_msft_onnx"
 
from dynamic_reconfigure.parameter_generator_catkin import *
import os

os.chdir('../../../')
path = os.getcwd()

newPath = path.replace(os.sep, '/') + "/src/ros_msft_onnx/ros_msft_onnx/testdata/"
gen = ParameterGenerator()
 
gen.add("tensor_height",    int_t,    0, "The height of the tensor", 416, 0, 3072)
gen.add("tensor_width",    int_t,    0, "The width of the tensor", 416, 0, 4096)
gen.add("confidence", double_t, 0, "The confidence parameter",    .5, 0, 1)
gen.add("onnx_model_path",    str_t,    0, "The path of the onnx model",  newPath + "model.onnx")
gen.add("onnx_label_path",    str_t,    0, "The path of the class labels",  newPath + "labels.txt")
gen.add("tracker_type",    str_t,    0, "Name of the tracker",  "yolo")
gen.add("image_processing",    str_t,    0, "Changes the image processing type",  "resize")
gen.add("input_node_name",    str_t,    0, "Name of the input for onnx model",  "image")
gen.add("output_node_name",    str_t,    0, "Name of the output for onnx model",  "grid")
gen.add("debug",   bool_t,   0, "Sets the debug value",  True)
gen.add("anch0", double_t, 0, "anchor0", 1.08, 0, 20)
gen.add("anch1", double_t, 0, "anchor1", 1.19, 0, 20)
gen.add("anch2", double_t, 0, "anchor2", 3.42, 0, 20)
gen.add("anch3", double_t, 0, "anchor3", 4.41, 0, 20)
gen.add("anch4", double_t, 0, "anchor4", 6.63, 0, 20)
gen.add("anch5", double_t, 0, "anchor5", 11.38, 0, 20)
gen.add("anch6", double_t, 0, "anchor6", 9.42, 0, 20)
gen.add("anch7", double_t, 0, "anchor7", 5.11, 0, 20)
gen.add("anch8", double_t, 0, "anchor8", 16.62, 0, 20)
gen.add("anch9", double_t, 0, "anchor9", 10.52, 0, 20)

 
exit(gen.generate(PACKAGE, "ros_msft_onnx", "reconfig"))